model_tests.rb

```diff
@@ -858,6 +859,205 @@ def autosized_fields(model, obj_types_to_skip, missing_getters)
+    end_year = dr_end.year
+
+    # Day/Month goes into Runperiod
+    r.setBeginDayOfMonth(start_day)
+    r.setBeginMonth(start_month)
+    r.setEndDayOfMonth(end_day)
+    r.setEndMonth(end_month)
+
+    # Year goes into YearDescription
+    yd.setCalendarYear(start_year)
+  else
+    # Defaults to full year, with assumed base year (2009)
+    start_year = yd.assumedYear
+    end_year = start_year
+
+    start_day = r.getBeginDayOfMonth
+    start_month = r.getBeginMonth
+    end_day = r.getEndDayOfMonth
+    end_month = r.getEndMonth
+
+    # Double check that we have the right defaults
+    assert_equal(1, start_day)
+    assert_equal(1, start_month)
+    assert_equal(31, end_day)
+    assert_equal(12, end_month)
+    assert_equal(2009, start_year)
+
+  end
+
+  # OpenStudio dates
+  d_start = OpenStudio::Date.new(OpenStudio::MonthOfYear.new(start_month),
+                                 start_day, start_year)
+  d_end = OpenStudio::Date.new(OpenStudio::MonthOfYear.new(end_month),
+                               end_day, end_year)
+
+
+
+  dir = File.join($TestDir, filename)
+  osw = File.join(dir, 'in.osw')
+  out_osw = File.join(dir, 'out.osw')
+  in_osm = File.join(dir, 'in.osm')
+  sql_path = File.join(dir, 'run', 'eplusout.sql')
+
+  # Run the workflow
+  # Switch this to to false if you're just modifying the code below
+  # (= the checks) after a successful first run as you don't have to wait
+  # for the simulation itself to rerun
+  run_sim = true
+  if run_sim
+    FileUtils.rm_rf(dir) if File.exists?(dir)
+    FileUtils.mkdir_p(dir)
+    # FileUtils.cp($OswFile, osw)
+
+    # Note: If you give E+ a weather file that doesn't have a leap day
+    # It won't give you an output that has a leap day...
+    # And it doesn't appear to let you know about it either
+    if options[:isLeapYear]
+      weather_file = "USA_IL_Chicago-OHare.Intl.AP.725300_AMY_2012_LeapYear.epw"
+    else
+      # If we have a start date, assume an AMY
+      if [:start]
+        weather_file = "USA_IL_Chicago-OHare.Intl.AP.725300_AMY_2012_NonLeapYear.epw"
+      else
+        # Otherwise a TMY
+        weather_file = "USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw"
+      end
+    end
+
+
+    osw_content = {
+      "weather_file": "../../weatherdata/#{weather_file}",
+      "seed_file": "in.osm",
+      # "steps": [],
+    }
+
+    File.write(osw, JSON.pretty_generate(osw_content))
+
+
+    # Save to above model into dir
+    m.save(in_osm, true)
+
+    # Run it
+    command = "\"#{$OpenstudioCli}\" run -w \"#{osw}\""
+    #command = "\"#{$OpenstudioCli}\" run --debug -w \"#{osw}\""
+
+    run_command(command, dir, 3600)
+  end
+
+  # Typically we do this:
+  # result_osw = postprocess_out_osw_and_copy(filename)
+  # Here we actually don't care about the OSW at all,
+  # just want to make sure that E+ completed successfully
+  # (though I don't see a reason it wouldn't)
+  fail "Cannot find file #{out_osw}" if !File.exists?(out_osw)
```

Comment:
@jmarrec seems like this could be factored into a post process routine similar to postprocess_out_osw_and_copy right?